app/components/category/category.service.js

```diff
@@ -0,0 +1,21 @@
+class CategoryService {
+    constructor($http, $state, appConfig) {
+        'ngInject';
+
+        this.$http = $http;
+        this.$state = $state;
+        this.appConfig = appConfig;
+    }
+
+    getCategories() {
+        return this.$http.get(this.appConfig + 'categories' + '.json')
+                .then((response) => {
+                return response.data;
+    })
+    .catch((response) => {
```

Comment:
Maybe create a function in service that you can put in catch block and control request failure in every of them?